pkg/provider/aws/resources/eks.go

```diff
@@ -135,23 +154,38 @@ func CreateEksCluster(cfg *config.Application, clusterName string, subnets []*Su
 		key := groupKey{InstanceType: params.InstanceType, NetworkType: unitCfg.NetworkPlacement}
 		spec := groupSpecs[key]
 		if spec == nil {
-			spec = &groupSpec{}
+			spec = &groupSpec{
+				DiskSizeGiB: 20,
+			}
 			groupSpecs[key] = spec
 		}
 		spec.refs = append(spec.refs, unit.AnnotationKey)
-		spec.DiskSizeGiB = int(math.Max(float64(spec.DiskSizeGiB), float64(spec.DiskSizeGiB)))
+		if params.DiskSizeGiB > spec.DiskSizeGiB {
+			spec.DiskSizeGiB = params.DiskSizeGiB
+		}
 	}
 
-	appName := cfg.AppName
+	var groups []*EksNodeGroup
 
-	clusterRole := createClusterAdminRole(appName, clusterName+"-k8sAdmin", references)
-	dag.AddResource(clusterRole)
-
-	cluster := NewEksCluster(appName, clusterName, subnets, securityGroups, clusterRole)
-	cluster.ConstructsRef = references
-	dag.AddDependenciesReflect(cluster)
+	hasInstanceType := false
+	for gk := range groupSpecs {
+		if gk.InstanceType != "" {
+			hasInstanceType = true
+			break
+		}
+	}
+	if !hasInstanceType {
+		for gk, spec := range groupSpecs {
+			if gk.InstanceType == "" {
+				groupSpecs[groupKey{InstanceType: "t3.medium", NetworkType: gk.NetworkType}] = spec
+			}
+		}
+	}
 
 	for groupKey, spec := range groupSpecs {
+		if groupKey.InstanceType == "" {
+			continue
+		}
```

Comment:
the issue with having this check down here is it practically ignores the DiskSize param on an exec unit if it doesnt set the instance type

Reply:
That's in parity with the logic before:
https://github.com/klothoplatform/klotho/blob/main/pkg/infra/pulumi_aws/iac/eks.ts#L479-L502

I can add a comment though for the future.